source/postprocess/visualization/fluid_velocity.cc

```diff
@@ -0,0 +1,177 @@
+                  const double porosity = std::max(in.composition[q][porosity_idx], 0.0);
+                  const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(in.position[q]);
+                  const double solid_density = out.densities[q];
+                  // const double fluid_viscosity = fluid_out->fluid_viscosities[q];
+                  // const double fluid_density = fluid_out->fluid_densities[q];
+                  // const double permeability = fluid_out->permeabilities[q];
+                  const double fluid_viscosity = 10;
+                  const double fluid_density = 1000;
+                  const double permeability = 1e-12;
+                  const Tensor<1,dim> fluid_velocity = solid_velocities[q] -
+                                                       permeability / fluid_viscosity / porosity * gravity *
+                                                       (solid_density - fluid_density) * velocity_scaling_factor;
+
+                  if (dim==2)
+                    {
+                      computed_quantities[q](0) =  fluid_velocity[0];
+                      computed_quantities[q](1) = fluid_velocity[1];
+                    }
+
+                  if (dim==3)
+                    {
+                      computed_quantities[q](0) =  fluid_velocity[0];
+                      computed_quantities[q](1) = fluid_velocity[1];
+                      computed_quantities[q](2) = fluid_velocity[2];
+                    }
+
+                }
+            }
+      }
+
+
+      template <int dim>
+      std::vector<std::string>
+      FluidVelocity<dim>::get_names () const
+      {
+        std::vector<std::string> names;
+        names.emplace_back("fluid_velocity_x");
+        names.emplace_back("fluid_velocity_y");
+        if (dim==3)
+          names.emplace_back("fluid_velocity_z");
+        return names;
+      }
+
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      FluidVelocity<dim>::get_data_component_interpretation () const
+      {
+        return
+          std::vector<DataComponentInterpretation::DataComponentInterpretation>
+          (dim,DataComponentInterpretation::component_is_scalar);
+      }
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(FluidVelocity,
+                                                  "fluid velocity",
+                                                  "A visualization output object that outputs the polar coordinates "
+                                                  "components $v_r$ and $v_\\phi$ of the velocity field in 2d and the "
+                                                  "spherical coordinates components $v_r$, $v_{\\phi}$ and $v_{\\theta}$ "
+                                                  "of the velocity field in 3d."
```

Comment:
This needs to be updated (but at least now I know where you copied everything from :D)